### DISCUSSION TITLE
你好，请教关于配置的问题

### DISCUSSION THREAD

**Discussion by changeche01** [swuecho/chat#92] [Q&A] [answered]
感谢大佬的分享。

请问是在哪里修改openai 模型 。
以及在哪里修改这些规则的配置。
谢谢！

> 第一个消息是系统消息（prompt）
> 上下文默认附带最新创建的10条消息
> 第一个注册的用户是管理员
> 默认限流 100 chatgpt api call /10分钟

**Comment by swuecho** [comment]
![Screenshot_20230324_205139_com android chrome](https://user-images.githubusercontent.com/666683/227526177-b999c21d-87f7-4512-835c-baa21e462a6a.jpg)

**Comment by swuecho** [comment]
模型这个参数我没添加 因为我觉得gpt3.5综合最优

**Comment by swuecho** [comment]
![Screenshot_20230324_205442_com android chrome](https://user-images.githubusercontent.com/666683/227526938-631bb898-5bdb-4c4f-b638-cd39475ae6f3.jpg)

**Comment by changeche01** [comment]
感谢大佬的快速回复。

发现有个问题， 就是设置最大token 回答数的配置， 我改为2000个 ，还是 很短的回答 只有512个。 不知道是哪里配置粗错了。

**Comment by swuecho** [comment]
> 感谢大佬的快速回复。
> 
> 发现有个问题， 就是设置最大token 回答数的配置， 我改为2000个 ，还是 很短的回答 只有512个。 不知道是哪里配置粗错了。

这个最大2000个不是一定2000个 你要长回答可以在prompt里面说 需要长答案 至少多少token

另外，这个token 参数我也不是很明白 需要改进

**Comment by swuecho** [comment]
> 感谢大佬的快速回复。
> 
> 发现有个问题， 就是设置最大token 回答数的配置， 我改为2000个 ，还是 很短的回答 只有512个。 不知道是哪里配置粗错了。

还有个可能我把超时限制的太短了, 导致出了bug.

**Comment by changeche01** [comment]
我找到了设置 token 数无效的原因，我找到必现的场景。

我题词   “写一个 vue hello world”  会发现被截断， 一开始我以为是token 字数不对，  后来多试几次 ，都是在同一个地方截断。

我觉得应该是 字符串 转义 处理的问题，导致 会话提前结束 或者不打印出来。

另外， 大佬你的后端是go 怎么编译的，放到docker 中的哪个文件， 我看看怎么改一下。

错误请见

![1](https://user-images.githubusercontent.com/127863646/227712093-719b9ff0-65cd-4d50-af9a-1682511518a1.jpg)

**Comment by swuecho** [comment]
> 还有个可能我把超时限制的太短了, 导致出了bug.

还有个可能我把超时限制的太短了, 导致出了bug.

这个已经修了, 我昨天也发现不对劲了. 

你更新一下最新image 就好了

**Comment by swuecho** [comment]
https://github.com/swuecho/chat/blob/master/.github/workflows/docker-image.yml

这个是整个项目的构建流程

**Comment by changeche01** [comment]
的确 更新了，就没有异常截断的问题。 
谢谢大佬，
我继续测试看有没有其他问题。

**Comment by changeche01** [comment]
大佬，你好，

我觉得还是会存在字符串 处理和 显示的问题。

提问  hello world vue
必定会在 
```
  <script>
    var app = new Vue({
      el: '#app',
```
这个地方停止显示，必现。 

然后需要重新刷新一下页面， 内容才会显示全。

```
  <script>
    var app = new Vue({
      el: '#app',
      data: {
        greeting: 'Hello',
        message: 'World!'
      }
    })
  </script>
```

请大佬帮忙看一下。

**Comment by swuecho** [comment]
> hello world vue

我大致知道哪里的问题了, 暂时刷新来解决.

**Comment by swuecho** [comment]
`data: {"id":"6yZjq1qR5g2rvvKMcEO4MaIvFw3kL","object":"","created":0,"model":"","choices":[{"index":0,"delta":{"content":"Here is an example code snippet for a \"Hello World\" message using Vue:\n\nHTML:\n```\n\u003cdiv id=\"app\"\u003e\n    \u003cp\u003e{{ message }}\u003c/p\u003e\n\u003c/div\u003e\n```\n\nJavaScript:\n```\nvar app = new Vue({\n    el: '#app',\n    data: {\n        message: 'Hello World!'\n    }\n});\n```\n\nThis code creates a new Vue instance that binds the `message` data property to the DOM element with ID `app`. The expression `{{ message }}` in the HTML template will display the value of `message`, which is \"Hello World!\". When you run this code, you should see a \"Hello World!\" message displayed on the webpage."},"finish_reason":""}]}`

**Comment by swuecho** [comment, marked as answer]
代码中的 data:  与 responseText 的data: 解析的时候混淆了.

```
-            const lastIndex = responseText.lastIndexOf('data: ')
+            const lastIndex = responseText.lastIndexOf('\n\ndata:')
             // Extract the JSON data chunk from the responseText
-            const chunk = responseText.slice(lastIndex + 6)
-
+            const chunk = responseText.slice(lastIndex + 8)
```

**Comment by changeche01** [comment]
对对， 我也想到是  \ndata    这里出错了。
